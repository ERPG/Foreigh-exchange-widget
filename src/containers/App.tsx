import { useState } from 'react';
import { Paper, Switch } from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';
import { darkTheme, lightTheme, WrapperTheme } from '../common/styles/theme';

import ExchangeWrapper from './ExchangeWrapper/ExchangeWrapper';
import { PaletteType } from '@material-ui/core';
import { Box } from '@mui/system';

const App = () => {
  const [themeMode, setThemeMode] = useState<PaletteType>(WrapperTheme.light);
  const [checked, handleCheck] = useState<boolean>(false);

  const customTheme = themeMode === WrapperTheme.light ? lightTheme : darkTheme;

  const handleThemeSwitchChange = (): void => {
    const activeTheme = !checked ? WrapperTheme.light : WrapperTheme.dark;
    handleCheck(!checked);
    setThemeMode(activeTheme);
  };

  return (
    <ThemeProvider theme={customTheme}>
      <Box>
        <Paper
          style={{
            height: 600,
            marginTop: 60,
            textAlign: 'center',
            margin: '0 auto',
            width: 300,
          }}
          elevation={1}
        >
          <Switch
            checked={checked}
            onChange={handleThemeSwitchChange}
            inputProps={{ 'aria-label': 'controlled' }}
          />
          <ExchangeWrapper />
        </Paper>
      </Box>
    </ThemeProvider>
  );
};

export default App;
