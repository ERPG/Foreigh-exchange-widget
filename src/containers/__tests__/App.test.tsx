

import { generateAppStateMock } from '../../../__tests__/__fixtures__/generateAppStateMock';
import { renderStoreHelper } from '../../../__tests__/__utils__/renderStoreHelper';
import App from '../App';

describe('App component', () => {
  it('Should render component without errors', () => {
    const { container } = renderStoreHelper({
      component: <App />,
      rootState: generateAppStateMock(),
    });

    expect(container).toMatchSnapshot();
  });
});
