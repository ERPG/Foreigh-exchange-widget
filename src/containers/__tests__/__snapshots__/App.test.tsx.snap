// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render component without errors 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      style="height: 600px; text-align: center; margin: 0px auto; width: 300px;"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-y18msa-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            aria-label="controlled"
            class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <div
        style="margin-top: 80px;"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
          style="text-align: center; height: 80px; line-height: 60px; margin: 16px; display: flex; justify-content: space-around; align-items: center;"
        >
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
                data-shrink="true"
                for="currency-amount"
                id="currency-amount-label"
              >
                Sending
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  data-fx="fx-amount-input"
                  id="currency-amount"
                  inputmode="numeric"
                  name="baseRate-input"
                  type="text"
                  value="0"
                />
              </div>
            </div>
          </div>
          <div
            style="margin-top: 16px;"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-1v3hyuc-MuiFormControl-root"
            >
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl makeStyles-root-1 MuiSelect-root css-1xuetnh-MuiInputBase-root-MuiInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-label="Without label"
                  aria-labelledby="mui-component-select-baseRate-select"
                  class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-a68jxg-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-baseRate-select"
                  role="button"
                  tabindex="0"
                >
                  EUR
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  data-fx="fx-currency-selection"
                  name="baseRate-select"
                  tabindex="-1"
                  value="1"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-1g3ozia-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
          style="text-align: center; height: 80px; line-height: 60px; margin: 16px; display: flex; justify-content: space-around; align-items: center;"
        >
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiFormLabel-filled"
                data-shrink="true"
                for="currency-amount"
                id="currency-amount-label"
              >
                Receiving
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                  data-fx="fx-amount-input"
                  id="currency-amount"
                  inputmode="numeric"
                  name="currentRate-input"
                  type="text"
                  value="0"
                />
              </div>
            </div>
          </div>
          <div
            style="margin-top: 16px;"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-1v3hyuc-MuiFormControl-root"
            >
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl makeStyles-root-1 MuiSelect-root css-1xuetnh-MuiInputBase-root-MuiInput-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-label="Without label"
                  aria-labelledby="mui-component-select-currentRate-select"
                  class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-a68jxg-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-currentRate-select"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  data-fx="fx-currency-selection"
                  name="currentRate-select"
                  tabindex="-1"
                  value="0"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-1g3ozia-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          style="margin-top: 80px; padding: 16px;"
        >
          <button
            class="makeStyles-root-2 MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root css-1gan90n-MuiButtonBase-root-MuiButton-root"
            data-fx="fx-button"
            name="exchange-button"
            tabindex="0"
            type="button"
          >
            <span>
              Send
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
